local/convergence.go

```diff
@@ -102,6 +111,28 @@ func (s *local) ensureService(ctx context.Context, project *types.Project, servi
 	return eg.Wait()
 }
 
+func (s *local) waitDependencies(ctx context.Context, project *types.Project, service types.ServiceConfig) error {
+	eg, ctx := errgroup.WithContext(ctx)
+	for dep, config := range service.DependsOn {
+		switch config.Condition {
+		case "service_healthy":
```

Comment:
Intuitively, I feel this waiting code could be integrated in `inDependencyOrder()`, and that would make things easier to understand (I hope). Here the complexity of the scheduling is split in 2 places and it makes it hard to understand the possible interactions between the 2. (Also note in terms of runtime, the function passed to `inDependencyOrder()` will precisely first call `waitDependencies`). 

Or possibly start services more brutally, and add a case here for dependencies, not only waiting on healthchecks, leaving each service the responsibility to wait for all it's dependencies in the proper state (requires refactoring to share status of all services)

Reply:
I would need to try things out, and discuss this more in detail...

Reply:
`inDependencyOrder` only sort services so we process services in the right order
service healthy handle the _condition_ a depdency is considered fullfiled and we can process the next step

> leaving each service the responsibility to wait for all it's dependencies in the proper state
this is what swarm decided to adopt, and cause for depends_on to be removed from compose file v3